test/e2e/BondingCurve.t.sol

```diff
@@ -0,0 +1,138 @@
+    address bob = address(0x606);
+
+    ERC20Mock token = new ERC20Mock("Mock", "MOCK");
+
+    function test_e2e_OrchestratorFundManagement() public {
+        // address(this) creates a new orchestrator.
+        IOrchestratorFactory.OrchestratorConfig memory orchestratorConfig =
+        IOrchestratorFactory.OrchestratorConfig({
+            owner: address(this),
+            token: token
+        });
+
+        IOrchestrator orchestrator =
+        _createNewOrchestratorWithAllModules_withBondingCurveFundingManager(
+            orchestratorConfig
+        );
+
+        BancorVirtualSupplyBondingCurveFundingManager fundingManager =
+        BancorVirtualSupplyBondingCurveFundingManager(
+            address(orchestrator.fundingManager())
+        );
+
+        // IMPORTANT
+        // =========
+        // Due to how the underlying rebase mechanism works, it is necessary
+        // to always have some amount of tokens in the fundingmanager.
+        // It's best, if the owner deposits them right after deployment.
+        //uint initialDeposit = 10e18;
+        //token.mint(address(this), initialDeposit);
+        //token.approve(address(fundingManager), initialDeposit);
+        //fundingManager.deposit(initialDeposit);
+
+        // Mint some tokens to alice and bob in order to fund the fundingmanager.
+        token.mint(alice, 1000e18);
+        token.mint(bob, 5000e18);
+
+        // Alice funds the fundingmanager with 1k tokens.
+        vm.startPrank(alice);
+        {
+            // Approve tokens to orchestrator.
+            token.approve(address(fundingManager), 1000e18);
+
+            // Deposit tokens, i.e. fund the fundingmanager.
+            fundingManager.buyOrder(1000e18);
+
+            // After the deposit, alice received some amount of receipt tokens
+            // from the fundingmanager.
+            assertTrue(fundingManager.balanceOf(alice) > 0);
+        }
+        vm.stopPrank();
+
+        // Bob funds the fundingmanager with 5k tokens.
+        vm.startPrank(bob);
+        {
+            // Approve tokens to fundingmanager.
+            token.approve(address(fundingManager), 5000e18);
+
+            // Deposit tokens, i.e. fund the fundingmanager.
+            fundingManager.buyOrder(5000e18);
+
+            // After the deposit, bob received some amount of receipt tokens
+            // from the fundingmanager.
+            assertTrue(fundingManager.balanceOf(bob) > 0);
+        }
+        vm.stopPrank();
+
+        // If the orchestrator spends half of the deposited tokens in the fundingmanager, i.e. for a milestone,
+        // alice and bob are still able to withdraw their respective leftover
+        // of the tokens.
+        // Note that we simulate orchestrator spending by just burning tokens.
+        uint halfOfDeposit = token.balanceOf(address(fundingManager)) / 2;
+        fundingManager.setVirtualCollateralSupply(halfOfDeposit);
+        /*         token.burn(
```

Comment:
outcommented the burning of the tokens

Reply:
Forgot to remove the comment. Since we are using a virtual supply we don't need to actually burn anything.